### DISCUSSION TITLE
Release 0.5.9 (1.20.80.22) + Release 0.5.10 (1.20.80.23)

### DISCUSSION THREAD

**Discussion by m-brisch** [Mojang/minecraft-editor#168] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/167)_  

  

---  
  

## Summary  

Hello! Two-fer update summary! 0.5.9 (1.20.80.22) + 0.5.10 (1.20.80.23), as of Wednesday, March 20, 2024.  

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and  [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

 

---   

## New Feature: Dynamic Pane Layout
We added a new layout system with three panes:
- Tool Pane (left one): shows the current tool that you're using
- Log Pane (bottom one): dedicated area for what previously lived as the floating log window
- Global Pane (right one): dedicated area to show information and menus that's not tied to a single tool (settings/test-world/etc)

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/70872def-90d9-4a23-964e-f081e6321105)

You can click and drag to resize the panes, click the bar to manually show/hide, or hit the lock icon to make them automatically show/hide when you hover over them. Try it out and let us know what you think 🙂

## New Feature: Menu System Upgrade
We added a new set of APIs to enable Editor Extensions to add elements to the top level menu.

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/0ad4357a-0003-4de4-82d1-361398acd012)

Check out the updated `Minimal Extension` sample in the starter-kit to get started on how to use the new `getMenu` function:
https://github.com/Mojang/minecraft-editor-extension-starter-kit/blob/main/templates/minimal.zip

## Change Log   
- Added three classes: the IBlockPaletteItem interface and the ProbabilityBlockPaletteItem and SimpleBlockPaletteItem concrete classes. These classes will be used in the later implementation of the Block Palette
- Tool mode screen layout will now be using a fixed drawer system rather than floating window panels. Drawers panels support drag to resize, expand/collapse, and show/hide behaviors 
- ModalTool.bindPropertyPane API function will now only bind a single pane to the tool. Bound pane will be displayed in the layout drawer next to the tool rail on tool activation. It is recommended to update the pane state once it’s associated with the tool to ensure it’s displayed correctly 
- Changed top level menus to show up only when they contain items 
- Added uniqueId property to IMenuCreationParams API which will replace IMenu id on creation 
- Added menubar: IMenuContainer property to IPlayerUISession. IMenuContainer will manage menu related operations 
- Added getMenu method to IMenuContainer to return an existing menu item. Predefined editor menus can be retrieved with the new CoreMenuType enum 
- Moved createMenu from IPlayerUISession to IMenuContainer
- Hovering over tool rail will now expand modal tool drawer if it’s not locked
- Updated Quick Start panel information
 
---   

  

Thank you for being here! You're the best! ❤️

**Comment by EnvisagedMyron** [comment]
where am i
